src/NHibernate/Proxy/Poco/BasicLazyInitializer.cs

```diff
@@ -14,22 +14,73 @@ public abstract class BasicLazyInitializer : AbstractLazyInitializer
 	{
 		private static readonly IEqualityComparer IdentityEqualityComparer = new IdentityEqualityComparer();
 
-		internal System.Type persistentClass;
+		// Since 5.1
+		[Obsolete("This field has no inherited usages in NHibernate and will be removed.")]
+		[NonSerialized]
 		protected internal MethodInfo getIdentifierMethod;
+		// Since 5.1
+		[Obsolete("This field has no inherited usages in NHibernate and will be removed.")]
+		[NonSerialized]
 		protected internal MethodInfo setIdentifierMethod;
+		// Since 5.1
+		[Obsolete("This field has no inherited usages in NHibernate and will be removed.")]
+		[NonSerialized]
 		protected internal bool overridesEquals;
+		// Since 5.1
+		[Obsolete("This field has no inherited usages in NHibernate and will be removed.")]
+		[NonSerialized]
 		protected internal IAbstractComponentType componentIdType;
 
+		[NonSerialized]
+		private System.Type _persistentClass;
+		private SerializableSystemType _serializablePersistentClass;
+		[NonSerialized]
+		private MethodInfo _getIdentifierMethod;
+		private SerializableMethodInfo _serializableGetIdentifierMethod;
+		[NonSerialized]
+		private MethodInfo _setIdentifierMethod;
+		private SerializableMethodInfo _serializableSetIdentifierMethod;
+		private readonly bool _overridesEquals;
+		private readonly IAbstractComponentType _componentIdType;
+
 		protected internal BasicLazyInitializer(string entityName, System.Type persistentClass, object id, 
 			MethodInfo getIdentifierMethod, MethodInfo setIdentifierMethod, 
 			IAbstractComponentType componentIdType, ISessionImplementor session, bool overridesEquals)
 			: base(entityName, id, session)
 		{
-			this.persistentClass = persistentClass;
+			_persistentClass = persistentClass;
+			_getIdentifierMethod = getIdentifierMethod;
+			_setIdentifierMethod = setIdentifierMethod;
+			_componentIdType = componentIdType;
+			_overridesEquals = overridesEquals;
+
+#pragma warning disable 618
 			this.getIdentifierMethod = getIdentifierMethod;
 			this.setIdentifierMethod = setIdentifierMethod;
 			this.componentIdType = componentIdType;
 			this.overridesEquals = overridesEquals;
+#pragma warning restore 618
+		}
+
+		[OnSerializing]
+		private void OnSerializing(StreamingContext context)
+		{
+			_serializablePersistentClass = SerializableSystemType.Wrap(_persistentClass);
+			_serializableGetIdentifierMethod = SerializableMethodInfo.Wrap(_getIdentifierMethod);
+			_serializableSetIdentifierMethod = SerializableMethodInfo.Wrap(_setIdentifierMethod);
+		}
+
+		[OnDeserialized]
+		private void OnDeserialized(StreamingContext context)
+		{
+			_persistentClass = _serializablePersistentClass?.GetSystemType();
+			_getIdentifierMethod = _serializableGetIdentifierMethod?.Value;
+			_setIdentifierMethod = _serializableSetIdentifierMethod?.Value;
+
+#pragma warning disable 618
+			getIdentifierMethod = _getIdentifierMethod;
+			setIdentifierMethod = _setIdentifierMethod;
```

Comment:
I was forgetting about re-hydrating the obsolete fields, and you are right doing it. But shouldn't the `componentIdType` and `overridesEquals` fields be re-hydrated here too?

Reply:
Yes they need re-hydrated as well, because `componentIdType` and `overridesEquals` are marked as NonSerialized.